.github/workflows/test.yml

```diff
@@ -1,36 +1,64 @@
 # Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
 # See the file LICENSE for licensing terms.
 
-name: Run Relayer Unit Tests
+name: Relayer Tests
 
 on:
-  push:
-    branches:
-      - '*'
+    push:
+        branches:
+            - "*"
 
 jobs:
-  build-test-relayer:
-    runs-on: ubuntu-20.04
-
-    steps:
-    - name: Checkout repository
-      uses: actions/checkout@v3
-      with:
-        path: awm-relayer
-
-    - name: Setup Go
-      uses: actions/setup-go@v4
-      with:
-        go-version: '1.20.7'
-
-    - name: Build Relayer
-      run: |
-        cd awm-relayer
-        go mod tidy
-        ./scripts/build.sh
-
-    - name: Run Relayer Unit Tests
-      run: |
-        cd awm-relayer
-        ./scripts/test.sh
+    build-test-relayer:
+        name: Build + Unit tests
+        runs-on: ubuntu-20.04
 
+        steps:
+            - name: Checkout repository
+              uses: actions/checkout@v3
+              with:
+                  path: awm-relayer
+
+            - name: Setup Go
+              uses: actions/setup-go@v4
+              with:
+                  go-version: "1.20.7"
+
+            - name: Build Relayer
+              run: |
+                  cd awm-relayer
+                  go mod tidy
+                  ./scripts/build.sh
+
+            - name: Run Relayer Unit Tests
+              run: |
+                  cd awm-relayer
+                  ./scripts/test.sh
+
+    e2e_tests:
+        runs-on: ubuntu-20.04
+        name: e2e_tests
+
+        steps:
+            - name: Checkout subnet-evm repository
+              uses: actions/checkout@v3
+              with:
+                  repository: ava-labs/subnet-evm
+                  ref: v0.5.4
+
+            - name: Setup Go
+              uses: actions/setup-go@v4
+              with:
+                  go-version: "1.20.7"
```

Comment:
We can set the go version as a global environment, rather than repeating it in the two jobs.